### PR TITLE
feat(dependabot): add GH action to auto assign reviewer to dependabot PRs

### PR DESCRIPTION
**Pull request by stephenliang**
This PR implements the ability for an action that executes on dependabot PRs to automatically assign the maintainer most applicable for the PR.

## Links

- [dependency update strategy](https://docs.google.com/document/d/1v9-Hq7Ybip83opfu9ABYueRp6voicdoN82n8Ex3gO64/edit)
- [test repo](https://github.com/code-dot-org/stephen-dependabot-autoassign-testing)
- [test repo actions](https://github.com/code-dot-org/stephen-dependabot-autoassign-testing/actions/workflows/assign-reviewer.yml)

## Testing story

Tested this by creating a `create-react-app` application in a test repo (see above) which has outdated dependencies. Upon the creation of a dependabot PR, the GH action will run and determine the applicable reviewer for the PR. Other PRs are ignored. Additionally, issuing a PR comment `/dependabot-reassign` retriggers the workflow in case it needs to be re-run for some reason.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: .github/workflows/assign-dependabot-reviewer.yml
Comment (cat5inthecradle) [inline .github/workflows/assign-dependabot-reviewer.yml]:
```suggestion
      - staging
```

File: .github/workflows/assign-dependabot-reviewer.yml
Comment (cat5inthecradle) [inline .github/workflows/assign-dependabot-reviewer.yml:9]: what's this for?

File: .github/workflows/assign-dependabot-reviewer.yml
Comment (cat5inthecradle) [inline .github/workflows/assign-dependabot-reviewer.yml]: Sucks that this is the only way to filter on author. This means the job will technically run on every PR, but at least it should fast-exit if the user doesn't match. It's noisy though.

Also noisy is my github action for [pinging infra on DMS Task edits](https://github.com/code-dot-org/code-dot-org/blob/a008c6e01cf236cc017c7a6ad60cb9165ad69391/.github/workflows/pr_check_for_manual_deploy_requirement.yml). I wonder if it would make sense to have a single complex action that handles this category of work? Probably premature, we can see how much we hate the noise.

File: .github/actions/assign-dependabot-reviewer/assign-reviewers.mjs
Comment (cat5inthecradle) [inline .github/actions/assign-dependabot-reviewer/assign-reviewers.mjs:116]: Should this be considered an error state? Should we notify a team that the reviewers map is outdated? Or is this better handled every couple of months via manual review?

**Review by cat5inthecradle — COMMENTED** [review]
Could we add some documentation here? Perhaps in the form of a comment that gets dropped onto the PR that points reviewers to our dependency maintenance plan and/or makes their options clear (merge this, defer this, ignore this, etc)

File: .github/workflows/assign-dependabot-reviewer.yml
Comment (stephenliang) [reply .github/workflows/assign-dependabot-reviewer.yml:9]: This allows the workflow to be restarted (in case there's a failure or some other reason) instead of waiting for dependabot to synchronize the PR. The comment would need to contain `/dependabot-reassign`

File: .github/workflows/assign-dependabot-reviewer.yml
Comment (stephenliang) [reply .github/workflows/assign-dependabot-reviewer.yml]: Since this is using a YAML if conditional, the workflow should be skipped completely if the PR wasn't opened by dependabot. I tested this by opening a PR and no action fires. However, if you issue the PR comment `/dependabot-reassign` it will trigger an action execution but it will be skipped after 3 seconds due to the conditional

Comment (cat5inthecradle) [reply .github/workflows/assign-dependabot-reviewer.yml]: oh, neat!

**Review by cat5inthecradle — APPROVED** [review]

File: .github/actions/assign-dependabot-reviewer/assign-reviewers.mjs
Comment (stephenliang) [reply .github/actions/assign-dependabot-reviewer/assign-reviewers.mjs:116]: No - this shouldn't happen as platform will be the default maintainer as a fallback. The log message indicates the fallback to platform failed and to avoid having the PR show a "failed" check (failure to assign the PR shouldn't block the PR), I have this silently fail instead.

**Comment by stephenliang** [comment]
> Could we add some documentation here? Perhaps in the form of a comment that gets dropped onto the PR that points reviewers to our dependency maintenance plan and/or makes their options clear (merge this, defer this, ignore this, etc)

Yes! I've added a new comment feature 

image

**Pull request merged**
